cmd/wallet-web/src/pages/mixins/common/helper.js

```diff
@@ -89,6 +89,44 @@ export const isVPType = (type) => toLower(type) == 'verifiablepresentation';
 export const getCredentialType = (types) =>
   types.filter((type) => type != 'VerifiableCredential')[0];
 
+// function to replace json path from the json objects
+export function getCredentialDisplayData(vc, manifest) {
+  // title
+  const titleFound = findByPath(vc, manifest.title.path);
+  const title = titleFound ? titleFound : manifest.title.fallback;
+  // icon
+  const icon = manifest.icon;
+
+  // find properties
+  const _readProperty = (property) => {
+    const pKey = property[0];
+    const value = findByPath(vc, property[1].path) || '';
+    const label = property.label;
+    const type = property.type;
+    const format = property.format;
+    return {
+      [pKey]: {
+        label,
+        value,
+        type,
+        format,
+      },
+    };
+  };
+
+  const properties = Object.entries(manifest.properties).map(_readProperty);
+  return {
+    title,
+    icon,
+    properties,
+  };
+}
+// function to replace json path from the json objects
+function findByPath(vc, ...path) {
+  // returns the first value from nested path.
+  return jp.query(vc, path[0][0])[0];
```

Comment:
Since we only update one string, we should just use `value` instead of `query` - it will update that value and return it, instead of returning an array